patches/minecraft/net/minecraft/util/math/RayTraceResult.java.patch

```diff
@@ -13,3 +13,29 @@
      private BlockPos field_178783_e;
      public RayTraceResult.Type field_72313_a;
      public EnumFacing field_178784_b;
+     public Vec3d field_72307_f;
+     public Entity field_72308_g;
++    public net.minecraftforge.fmp.multipart.IMultipart partHit;
++    public AxisAlignedBB boxHit;
+ 
+     public RayTraceResult(Vec3d p_i47094_1_, EnumFacing p_i47094_2_, BlockPos p_i47094_3_)
+     {
+@@ -48,13 +56,15 @@
+ 
+     public String toString()
+     {
+-        return "HitResult{type=" + this.field_72313_a + ", blockpos=" + this.field_178783_e + ", f=" + this.field_178784_b + ", pos=" + this.field_72307_f + ", entity=" + this.field_72308_g + '}';
++        return "HitResult{type=" + this.field_72313_a + ", blockpos=" + this.field_178783_e + ", f=" + this.field_178784_b + ", pos=" + this.field_72307_f + ", entity=" + this.field_72308_g + ", multipart=" + this.partHit + ", aabb=" + this.boxHit + '}';
+     }
+ 
+     public static enum Type
+     {
+         MISS,
+-        BLOCK,
+-        ENTITY;
++        BLOCK{ public boolean isBlock() { return true; } },
++        ENTITY,
++        MULTIPART{ public boolean isBlock() { return true; } };
++        public boolean isBlock() { return false; }
+     }
```

Comment:
I'd like to suggest reducing patch size with
`public boolean isBlock(){return this == BLOCK || this == MULTIPART;}`